app/src/main/java/com/github/sdp_begreen/begreen/fragments/UserPhotosViewAdapter.kt

```diff
@@ -42,43 +47,43 @@ class UserPhotosViewAdapter(
 
     }
 
+    @SuppressLint("SetTextI18n")
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
 
-        val photo = photos?.get(position)
+        // Prefer this semantic instead of the .let { } so we don't have to embed too much blocks
 
-        if(isFeed) {
+        val photos = photos ?: return
+        val photo = photos[position]
+        val userId = photo.takenBy ?: return
 
-            // Display avatar if on feed
-            val drawable = ContextCompat.getDrawable(holder.avatarView.context, R.drawable.ic_baseline_person)
-            val defaultAvatar = drawable?.toBitmap()
-            // holder.avatarView.setImageBitmap(getFromDB(photo) ?: defaultAvatar)
-            holder.avatarView.setImageBitmap( defaultAvatar)
+        lifecycleScope.launch {
 
-        }else{
-            // Do not display avatar if not on feed
-            holder.avatarView.visibility = View.GONE
-        }
+            val user = db.getUser(userId) ?: return@launch
 
-        if (photo is TrashPhotoMetadata) {
-            // use the photo variable as a TrashPhotoMetadata instance
-
-            //Set default value
-            holder.titleView.text = photo?.caption ?: "No title"
-            holder.subtitleView.text = (photo?.takenOn?.toString()
-                ?: "Unknown date") + " | " + (photo?.trashCategory?.title
-                ?: "No category")
-            //holder.photoView.setImageBitmap(photo.getPhotoFromDataBase())
-            //TODO------------FOR DEMO -----------------
-            val url = URL("https://picsum.photos/400")
-            holder.photoView.setImageBitmap(
-                BitmapFactory.decodeStream(
-                    url.openConnection().getInputStream()
-                )
-            )
-            //------------FOR DEMO -----------------
+            if (isFeed) {
+
+                holder.avatarView.visibility = View.VISIBLE
+
+                // Display avatar if on feed
+                user.profilePictureMetadata?.also {
+                    val avatarImage = db.getUserProfilePicture(it, userId)
+                    holder.avatarView.setImageBitmap(avatarImage)
+                }
+
+            } else {
+
+                // Do not display avatar if not on feed
+                holder.avatarView.visibility = View.GONE
+            }
+
+            // Display post content
+            val dateString = (photo.takenOn?.toString() ?: resources.getString(R.string.unknown_date))
+            val categoryString = (photo.trashCategory?.title ?: resources.getString(R.string.no_category))
 
-            // TODO : to change
-            holder.descriptionView.text = photo?.caption
+            holder.titleView.text = user.displayName ?: resources.getString(R.string.unknown_user)
+            holder.subtitleView.text = "$dateString | $categoryString"
```

Comment:
Here I would use a resource string with format. See this [link](https://stackoverflow.com/questions/12627457/format-statement-in-a-string-resource-file) for an example